### PR TITLE
Added various file formats to constants.py

### PR DESCRIPTION
**Pull request by PossiblePanda**
I added various file formats to the constants.py file, including ".dmg", ".s7z", ".cr2", ".jar", ".msi"

### CODE REVIEW

**Comment by PossiblePanda** [comment]
I just now realize that some of my additions are debatable, such as .jar could also be an archive file

**Comment by CyanVoxel** [comment]
".cr2" would be valuable to have, however both the `ARCHIVE_TYPES` and `PROGRAM_TYPES` lists aren't currently being used anywhere. There will be a shift towards using [MIME types](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types) in favor of these hardcoded lists in the future, however in the meantime it would probably be good to have at least the ".cr2" and ".s7z" extensions added. You're right with the ".jar" ambiguity, in addition to ".dmg" being more closely a disk image type and ".msi" also being in a nebulous space between a program and specifically an installer.

File: tagstudio/src/core/constants.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
     ".j2k",
     ".jpg2",
 ]
-RAW_IMAGE_TYPES: list[str] = [".raw", ".dng", ".rw2", ".nef", ".arw", ".crw", ".cr3"]
+RAW_IMAGE_TYPES: list[str] = [".raw", ".dng", ".rw2", ".nef", ".arw", ".crw", ".cr3", ".cr2"]
```
Comment (CyanVoxel) [inline tagstudio/src/core/constants.py:32]: Ruff formatting
```suggestion
RAW_IMAGE_TYPES: list[str] = [
    ".raw",
    ".dng",
    ".rw2",
    ".nef",
    ".arw",
    ".crw",
    ".cr2",
    ".cr3",
]
```

File: tagstudio/src/core/constants.py [hunk 2]
```diff
@@ -86,8 +86,8 @@
 ]
 SPREADSHEET_TYPES: list[str] = [".csv", ".xls", ".xlsx", ".numbers", ".ods"]
 PRESENTATION_TYPES: list[str] = [".ppt", ".pptx", ".key", ".odp"]
-ARCHIVE_TYPES: list[str] = [".zip", ".rar", ".tar", ".tar", ".gz", ".tgz", ".7z"]
-PROGRAM_TYPES: list[str] = [".exe", ".app"]
+ARCHIVE_TYPES: list[str] = [".zip", ".rar", ".tar", ".tar", ".gz", ".tgz", ".7z", ".s7z", ".dmg"]
+PROGRAM_TYPES: list[str] = [".exe", ".app", ".jar", ".msi"]
```
Comment (CyanVoxel) [inline tagstudio/src/core/constants.py:90]: Mentioned changes + Ruff formatting
```suggestion
ARCHIVE_TYPES: list[str] = [
    ".zip",
    ".rar",
    ".tar",
    ".tar",
    ".gz",
    ".tgz",
    ".7z",
    ".s7z",
]
PROGRAM_TYPES: list[str] = [".exe", ".app"]
```

**Review by CyanVoxel — CHANGES_REQUESTED** [review]

**Comment by PossiblePanda** [comment]
I think I accepted your commits correctly @CyanVoxel , not the best at doing PR's though so let me know if I did it wrong xD

**Comment by CyanVoxel** [comment]
> I think I accepted your commits correctly @CyanVoxel , not the best at doing PR's though so let me know if I did it wrong xD

You did excellent! And now you've got some more PR practice for the future 😉
Thank you for your work on this!

**Review by CyanVoxel — APPROVED** [review]

**Pull request merged**